File: socialview-commons/src/main/java/io/github/hendraanggrian/socialview/commons/HashtagAdapter.java
package io.github.hendraanggrian.socialview.commons;

import android.content.Context;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.text.NumberFormat;
import java.util.Locale;

/**
 * @author Hendra Anggrian (hendraanggrian@gmail.com)
 */
public final class HashtagAdapter extends SuggestionAdapter<Hashtag> {

    public HashtagAdapter(@NonNull Context context) {
        super(context, R.layout.item_hashtag, R.id.textview_hashtag_value);
    }

    @NonNull
    @Override
    public View getView(int position, View convertView, @NonNull ViewGroup parent) {
        final ViewHolder holder;
        if (convertView == null) {
            convertView = LayoutInflater.from(getContext()).inflate(R.layout.item_hashtag, parent, false);
            holder = new ViewHolder(convertView);
            convertView.setTag(holder);
        } else {
            holder = (ViewHolder) convertView.getTag();
        }

        final Hashtag item = getItem(position);
        if (item != null) {
            holder.textViewHashtag.setText(item.getHashtag());

            if (item.getHashtagCount() != null) {
                holder.textViewHashtagCount.setText(item.getHashtagCount() == 1
                        ? "1 post"
                        : String.format("%s posts", NumberFormat.getNumberInstance(Locale.US).format(item.getHashtagCount())));
                holder.textViewHashtagCount.setVisibility(View.VISIBLE);
            } else {
                holder.textViewHashtagCount.setVisibility(View.GONE);
            }
        }
        return convertView;
    }

    @NonNull
    @Override
    public SuggestionFilter initializeFilter() {
        return new SuggestionFilter() {
            @Override
            public String getString(Hashtag item) {
                return item.getHashtag();
            }
        };
    }

    private static class ViewHolder {
        private final TextView textViewHashtag, textViewHashtagCount;

        private ViewHolder(@NonNull View view) {
            textViewHashtag = (TextView) view.findViewById(R.id.textview_hashtag_value);
            textViewHashtagCount = (TextView) view.findViewById(R.id.textview_hashtag_count);
        }
    }
}

File: socialview-commons/src/com/hendraanggrian/appcompat/widget/HashtagAdapter.kt
package com.hendraanggrian.appcompat.widget

import android.content.Context
import android.support.annotation.PluralsRes
import android.view.LayoutInflater.from
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.hendraanggrian.appcompat.socialview.commons.R

/**
 * Default adapter for displaying hashtag in [SocialAutoCompleteTextView].
 * Note that this adapter is completely optional, any adapter extending
 * [android.widget.ArrayAdapter] can be attached to [SocialAutoCompleteTextView].
 */
class HashtagAdapter @JvmOverloads constructor(
    context: Context,
    @PluralsRes private val countPlural: Int = R.plurals.posts
) : SocialAdapter<Hashtag>(
    context,
    R.layout.socialview_layout_hashtag,
    R.id.socialview_hashtag
) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val holder: ViewHolder
        var view = convertView
        when (view) {
            null -> {
                view = from(context).inflate(R.layout.socialview_layout_hashtag, parent, false)
                holder = ViewHolder(view!!)
                view.tag = holder
            }
            else -> holder = view.tag as ViewHolder
        }
        getItem(position)?.let { hashtag ->
            holder.hashtagView.text = hashtag.hashtag
            hashtag.count?.let {
                holder.countView.text = context.resources.getQuantityString(countPlural, it)
            }
        }
        return view
    }

    private class ViewHolder(itemView: View) {
        val hashtagView: TextView = itemView.findViewById(R.id.socialview_hashtag)
        val countView: TextView = itemView.findViewById(R.id.socialview_hashtag_count)
    }
}